src/Runner.Worker/Container/ContainerHooks/ContainerHookInput.cs

```diff
@@ -0,0 +1,50 @@
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+
+namespace GitHub.Runner.Worker.Container.ContainerHooks
+{
+    public class HookInput
+    {
+        public HookCommand Command { get; set; }
+        public string ResponseFile { get; set; }
+        public HookArgs Args { get; set; }
+        public dynamic State { get; set; }
+    }
+
+    [JsonConverter(typeof(StringEnumConverter))]
+    public enum HookCommand
+    {
+        [EnumMember(Value = "prepare_job")]
+        PrepareJob,
+        [EnumMember(Value = "cleanup_job")]
+        CleanupJob,
+    }
+    public class HookArgs
```

Comment:
I think we should eventually just have a generic

`HookArgs` interface, which `PrepareJobArgs`, `CleanupJobArgs`, ect implement.